snuba/datasets/events.py

```diff
@@ -194,3 +201,12 @@ def get_promoted_context_columns(self):
 
     def get_required_columns(self):
         return self.__required_columns
+
+
+class EventsProcessor(MessageProcessor):
```

Comment:
The idea is to move all processor.py logic into this class and move it out of this file since that is event specific.
That will allow us to remove a lot of the cumbersome places where we pass datasets or column lists around that I introduced in the previous PRs